Todo/TaskListsViewController.swift

```diff
@@ -0,0 +1,271 @@
+    }
+    
+    // MARK: - UISearchController
+    
+    func updateSearchResultsForSearchController(searchController: UISearchController) {
+        let text = searchController.searchBar.text ?? ""
+        let active = !text.isEmpty
+        listsLiveQuery.startKey = active ? text : nil
+        listsLiveQuery.endKey = listsLiveQuery.startKey
+        listsLiveQuery.prefixMatchLevel = active ? 1 : 0
+        listsLiveQuery.queryOptionsChanged()
+    }
+    
+    // MARK: - Action
+    @IBAction func addAction(sender: AnyObject) {
+        Ui.showTextInputDialog(
+            onController: self,
+            withTitle: "New Task List",
+            withMessage:  nil,
+            withTextFieldConfig: { (textField) -> Void in
+                textField.placeholder = "List name"
+                textField.autocapitalizationType = .Words
+            },
+            onOk: { (name) -> Void in
+                self.createTaskList(name)
+            }
+        )
+    }
+    
+    @IBAction func logoutAction(sender: AnyObject) {
+        let app = UIApplication.sharedApplication().delegate as! AppDelegate
+        app.logout()
+    }
+
+    // MARK: - Database
+
+    func setupViewAndQuery() {
+        let listsView = database.viewNamed("list/listsByName")
+        if listsView.mapBlock == nil {
+            listsView.setMapBlock({ (doc, emit) -> Void in
+                if let type: String = doc["type"] as? String where type == "task-list" {
+                    if let name = doc["name"] {
+                        emit(name, nil)
+                    }
+                }
+            }, version: "1.0")
+        }
+
+        listsLiveQuery = listsView.createQuery().asLiveQuery()
+        listsLiveQuery.addObserver(self, forKeyPath: "rows", options: .New, context: nil)
+        listsLiveQuery.start()
+        
+        let incompTasksCountView = database.viewNamed("list/incompleteTasksCount")
+        if incompTasksCountView.mapBlock == nil {
+            incompTasksCountView.setMapBlock({ (doc, emit) -> Void in
+                if let type: String = doc["type"] as? String where type == "task" {
+                    if let list = doc["taskList"] as? [String: AnyObject] {
+                        if let listId = list["id"] {
+                            if !(doc["complete"] as? Bool ?? false) {
+                                emit(listId, nil)
+                            }
+                        }
+                    }
+                }
+            }, reduceBlock: { (keys, values, reredeuce) -> AnyObject in
+                return values.count
+            }, version: "1.0")
+        }
+        
+        incompTasksCountsLiveQuery = incompTasksCountView.createQuery().asLiveQuery()
+        incompTasksCountsLiveQuery.groupLevel = 1
+        incompTasksCountsLiveQuery.addObserver(self, forKeyPath: "rows", options: .New, context: nil)
+        incompTasksCountsLiveQuery.start()
+    }
+
+    func reloadTaskLists(queryEnum: CBLQueryEnumerator?) {
```

Comment:
It seems wrong to pass the enumerator as a parameter, because the only value that makes sense is `listsLiveQuery.rows`. Instead just use it directly.